### ISSUE TITLE
version 4.0.0 + grouped options = console error: "Warning: Encountered two children with the same key, `react-select-2-option-0-[object Object]"

### ISSUE THREAD

**Issue by mrudowski** [JedWatson/react-select#4395]
in 4.0.0 version when we click on select component to open it we get:

```
Warning: Encountered two children with the same key, `react-select-2-option-0-[object Object]`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
    in div (created by Group)
    in div (created by EmotionCssPropInternal)
    in EmotionCssPropInternal (created by Group)
    in Group (created by Select)
    in div (created by EmotionCssPropInternal)
    in EmotionCssPropInternal (created by MenuList)
    in MenuList (created by ScrollManager)
    in ScrollManager (created by MenuPlacer)
    in div (created by EmotionCssPropInternal)
    in EmotionCssPropInternal (created by Menu)
    in Menu (created by MenuPlacer)
    in MenuPlacer (created by Select)
    in div (created by EmotionCssPropInternal)
    in EmotionCssPropInternal (created by SelectContainer)
    in SelectContainer (created by Select)
    in Select (created by StateManager)
    in StateManager
    in Unknown
```

we can see it here (link from page - grouped example)
https://codesandbox.io/s/7utj1?module=/example

**Comment by ebonow** [comment]
Greetings @mrudowski ,

It is likely that this error was introduced in v3.2. I have identified it in part here: https://github.com/JedWatson/react-select/pull/4391

**Comment by Methuselah96** [comment]
@ebonow I think this might actually be a separate issue. Investigating it now.

**Comment by ebonow** [comment]
The culprit of that issue is that any prop that is not defined and filtered out by `GroupHeader` is applied as a DOM Attribute to the Component.

It may not be related, but it seemed familiar as `data=[object Object]` is being applied as an attribute to all GroupHeaders now.

**Comment by Methuselah96** [comment]
@ebonow Yeah, it definitely seems similar, but it's actually due to a different change. I've found the root issue and I'm working on a PR now.

**Comment by mrudowski** [comment]
You are really great people, all of you.
I admire all the work behind this fine component back from v1

**Comment by mrudowski** [comment]
oh, I'm sorry, I hoped we end here but I have further problem with grouped options 

**keyboard navigation (up/down arrow key) not working** - we can open menu but not move focus over the options

the same example (just change react-select version to 4.0.1)
https://codesandbox.io/s/7utj1?module=/example

Leave it here or open new issue?

**Comment by Methuselah96** [comment]
A new issue would be great for visibility, thanks! I’ll take a look at it
in a little bit.

**Comment by Methuselah96** [comment]
@mrudowski Should be resolved now in 4.0.2.

**Comment by mrudowski** [comment]
thank you
